### PR TITLE
Update project archetype to release 2.0.4 of the Core Components

### PR DESCRIPTION
**Pull request by richardhand**
- Updates core.wcm.components.all dependency to release 2.0.4
- Adds new proxy components for navigation, language navigation and search components
- Updates existing proxy component versions
- Updates clientlib versions
- Adds the navigation, language navigation and search components to the content page template structure
- Fixes bespoke breadcrumb styles for V2 breadcrumb component

### CODE REVIEW

File: src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/content-page/structure/.content.xml [hunk 1]
```diff
@@ -9,6 +9,15 @@
         <root
             jcr:primaryType="nt:unstructured"
             sling:resourceType="wcm/foundation/components/responsivegrid">
+            <languagenavigation
+                jcr:primaryType="nt:unstructured"
+                sling:resourceType="${appsFolderName}/components/content/languagenavigation"/>
+            <search
+                jcr:primaryType="nt:unstructured"
+                sling:resourceType="${appsFolderName}/components/content/search"/>
```
Comment (bpauli) [inline src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/content-page/structure/.content.xml]: Please add also mandatory searchRoot property to the template.

File: src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/content-page/structure/.content.xml [hunk 2]
```diff
@@ -9,6 +9,15 @@
...
+            <navigation
+                jcr:primaryType="nt:unstructured"
+                sling:resourceType="${appsFolderName}/components/content/navigation"/>
```
Comment (bpauli) [inline src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/templates/content-page/structure/.content.xml]: Also setting navigationRoot pointing to contentRoot would makes sense, or WDYT?
--

**Comment by richardhand** [comment]
Thanks @bpauli - I added some configurations in https://github.com/Adobe-Marketing-Cloud/aem-project-archetype/pull/110/commits/9c231ca9ca269feb5a82e2dfa2084e26a14f42c0

**Comment by richardhand** [comment]
Thanks @bpauli !

**Pull request merged**